### ISSUE TITLE
JavaScript

### ISSUE THREAD

**Issue by selrond** [wesbos/ama#72]
## Hey Wes!
I'd love to know, how you approached learning of JavaScript - I mean could you tell me a little about how you learned it? I started recently, but there's so much to learn plus I'm confused how to make it project-based because it feels like I can't do nothing "real" with vanilla JS except of pi-calculator-commad-line-program type of "projects"...

**Comment by Sivilll** [comment]
What the above is asking. I want some insight on this as well.. bits and pieces collected over a long time, is hard to put together.

**Comment by konekoya** [comment]
While waiting for his answer, check out this ama 
https://github.com/wesbos/ama/issues/37

**Comment by gethinoakes** [comment]
You guys could also check out [JS The Right Way](http://jstherightway.org/) and [Superhero JS](http://superherojs.com/), both pretty awesome resources and in a structured way for learning.

**Comment by Sivilll** [comment]
@gethinoakes Thank you.

**Comment by wesbos** [comment]
**TLDR: There aren't any shortcuts, put in the time. It gets easier and you get better as time goes on. You can expedite it by putting in the work and making lots of things.**

Long:

I'd love to say I read a few books, did a few key courses and BAM I was amazing but it was anything but that. More of a limp and a struggle over the years piecing things together and inching toward clarity. 

How did I learn? I made probably 2-3,000 things with JavaScript. I've built hundreds of websites and done hundreds of little code alongs myself. Whenever I needed a website to do something, I'd stop and figure it out. 

Seems pretty obvious, but it's all about grinding it out. Here are some tips that I use both to tell my students, but also when I create content that teaches things. 

1) Make sure you are excited about the project. If you are excited about it, you are invested in it and it gets a ton easier.  You will need to come to terms that you aren't building amazing applications right away - it's hard to see amazing parallax websites while you are still trying to make a rock paper scissors game

2) Pick a small, well defined application and build it. See it all that way through to the end. See it through the parts where it gets too hard and confusing. See it through the parts where you lose interest.  If you find it too hard, or uninteresting, you need to scale back the application to something that you can finish. 

Confidence in life comes from small wins and it's no different in programming.

3) Ask for help. Stack overflow. Slack Rooms. Meetups. You name it, don't be embarrassed to ask for help. I still do every day and it's the sign of a good programmer. 

4) Focus focus focus. Bootstap?! React? Angular! Foundation! Angular2! BAHhhhhhhhhhh

I'm not going to be the guy who says "Learn the basics first". While the basics are very important, it does get boring making simple apps all day long. Part of the reason I fell in love with JavaSCript is because jQuery allowed me to do such cool stuff without really knowing what I was doing. You obviously will go back in fill in the gaps as you learn.

**That said**, pick one and stick with it. They change so fast that you aren't helping yourself bouncing to and from framework-of-the-day. If you get really good at one thing, you'll realize that moving to another one really isn't all that hard. 


hope that help - feel free to ask followups

**Comment by selrond** [comment]
>Confidence in life comes from small wins and it's no different in programming.

Thank you @wesbos! You just helped me a lot!

**Comment by selrond** [comment]
@wesbos Could you list me at least 10 examples of simple yet entertaining projects to do in JS?

**Comment by gethinoakes** [comment]
@selrond https://github.com/tastejs/awesome-app-ideas

**Comment by selrond** [comment]
@gethinoakes this is gold! Thanks

**Comment by gethinoakes** [comment]
No problem @selrond :) I should actually start using that myself...
